File: app/src/main/java/com/example/imageviewer/ImageData.java
package com.example.imageviewer;

import android.net.Uri;

import java.io.Serializable;

public class ImageData implements Serializable {
    private Uri uri;
    private String name;
    public static String TAG = "ImageData";
    private int id;

    ImageData(Uri uri, int id) {
        this.uri = uri;
        name = uri.getLastPathSegment();
        this.id = id;
    }

    public String getUri() {
        return uri.toString();
    }

    public void setUri(String uriString) {
        this.uri = Uri.parse(uriString);
    }

    public void setName(String name){
        this.name = name;
    }

    public int getId(){
        return id;
    }

    public String getName() {
        return name;
    }
}

File: app/src/main/java/com/example/imageviewer/ImageData.kt
package com.example.imageviewer

import android.net.Uri
import java.io.Serializable

class ImageData(fullPhotoUri: Uri) : Serializable {
    var name: String
    val fileUriString: String

    val fileUri: Uri
        get() = Uri.parse(fileUriString)
    val fileName:String
        get() = name


    init {
        this.fileUriString = fullPhotoUri.toString()
        val path = fullPhotoUri.path!!.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        this.name = path[path.size - 1]
    }

}